### ISSUE TITLE
Possibly incompatible versions of react-router and react-router-config

### ISSUE THREAD

**Issue by GrigoryPtashko** [sibelius/relay-workshop#2699]
`yarn start` in the root gives this error in the browser:

Uncaught runtime errors:
×
ERROR
pathname.match is not a function. (In 'pathname.match(matcher)', 'pathname.match' is undefined)
matchPath@http://localhost:8444/main.js:21753:29
@http://localhost:8444/main.js:101624:85
some@[native code]
matchRoutes@http://localhost:8444/main.js:101623:14
matchRoute@http://localhost:8444/main.js:30712:89
createRouter@http://localhost:8444/main.js:30753:36
./src/client.tsx@http://localhost:8444/main.js:28316:77
@http://localhost:8444/main.js:140503:35
__webpack_require__@http://localhost:8444/main.js:139926:37
@http://localhost:8444/main.js:141150:56
global code@http://localhost:8444/main.js:141152:12

I guess that's because the signature of the `matchPath` changed from `matchPath(pathname, route)` to `matchPath(route, pathname)`. But even if I change this call directly in the `node_modules` I get in the browser:

Uncaught runtime errors:
×
ERROR
No route for /auth/login
matchRoute@http://localhost:8444/main.js:30714:20
@http://localhost:8444/main.js:30774:31
forEach@[native code]
call@http://localhost:8444/main.js:44343:23
applyTx@http://localhost:8444/main.js:43801:19
push@http://localhost:8444/main.js:43830:14
@http://localhost:8444/src_components_feed_Root_tsx.js:204:19
commitHookEffectListMount@http://localhost:8444/main.js:87426:32
commitPassiveMountOnFiber@http://localhost:8444/main.js:89202:38
commitPassiveMountEffects_complete@http://localhost:8444/main.js:89167:34
commitPassiveMountEffects_begin@http://localhost:8444/main.js:89154:41
commitPassiveMountEffects@http://localhost:8444/main.js:89142:34
flushPassiveEffectsImpl@http://localhost:8444/main.js:91315:28
flushPassiveEffects@http://localhost:8444/main.js:91260:37
@http://localhost:8444/main.js:91045:28
workLoop@http://localhost:8444/main.js:125951:42
flushWork@http://localhost:8444/main.js:125924:22
performWorkUntilDeadline@http://localhost:8444/main.js:126218:42

**Comment by sibelius** [comment]
I think we did not a full upgrade to the latest react-router v6

the fix would be to move everything to react router v6

**Comment by sibelius** [comment]
can you send a few pull request to fix all this problems ?

**Comment by GrigoryPtashko** [comment]
> can you send a few pull request to fix all this problems ?

Yes, sure. But it can take some time..

**Comment by sibelius** [comment]
you can send many pull requests, fixing one by one

**Comment by GrigoryPtashko** [comment]
Yes :) just give me some time, if it's possible. I want to contribute :)
